sed/binning/numba_bin.py

```diff
@@ -55,7 +55,11 @@ def _hist_from_bin_range(
         is_inside = True
         flatidx = 0
         for i in range(ndims):
-            j = (sample[t, i] - ranges[i, 0]) * delta[i]
+            # strip off numerical rounding errors
+            j = round((sample[t, i] - ranges[i, 0]) * delta[i], 11)
+            # add counts on last edge
+            if j == bins[i]:
```

Comment:
instead of comparing, what about something like this:
```
j = j - 0.5 * (j == bins[i])
```
the operation is then vectorized and we don't have to compare in every step

Reply:
We loop through each event, and each dimension individually (for t, for dim), so j is a single float number. I don't think that this will make a difference.

Reply:
Same as above.